pages/lobby_page.py

```diff
@@ -94,38 +109,26 @@ def send_invite(self):
+        return self.context.driver.find_element_by_xpath('//button[text()="Invite people"]')
 
     def invite(self):
         self.find_invite().click()
 
     def accept_invite(self):
-        if "chat" in self.context.driver.current_url:
-            room_xpath = '//div[contains(@class,"hc-lobby-panel-content")]//span[text()="'+LobbyPage.room_name+'"]'
-            self.context.driver.find_element_by_xpath(room_xpath).click()
-            self.context.driver.find_element_by_id('hc-message-input').send_keys('@all', Keys.RETURN, Keys.RETURN)
-            self.context.wait.until(EC.visibility_of_element_located((By.XPATH, '//*[@class="msg-line"]')))
-        elif "room" in self.context.driver.current_url:
-            self.context.driver.find_element_by_id('hc-message-input').send_keys('@all', Keys.RETURN, Keys.RETURN)
-            self.context.wait.until(EC.visibility_of_element_located((By.XPATH, '//*[@class="msg-line"]')))
+        self.context.driver.get(self.context.base_url + "/chat/room/" + url)
+        self.context.wait.until(lambda driver: driver.find_element_by_id('status_dropdown'))
+        self.context.driver.find_element_by_id('hc-message-input').send_keys('@all', Keys.RETURN, Keys.RETURN)
+        self.context.wait.until(EC.visibility_of_element_located((By.XPATH, '//*[@class="msg-line"]')))
 
     def delete_room(self):
-        if "chat" in self.context.driver.current_url:
-            room_xpath = '//div[contains(@class,"hc-lobby-panel-content")]//span[text()="'+LobbyPage.room_name+'"]'
-            self.context.driver.find_element_by_xpath(room_xpath).click()
-            self.context.driver.find_element_by_id('room-actions-btn').click()
-            self.context.driver.find_element_by_css_selector('.delete-room-action').click()
-            self.context.wait.until(EC.visibility_of_element_located((By.XPATH, '//button[text()="Delete room"]')))
-            self.context.driver.find_element_by_xpath('//button[text()="Delete room"]').click()
-            time.sleep(1)
-            # in theory this should help with problem "ghost" rooms,
-        elif "room" in self.context.driver.current_url:
-            self.context.driver.find_element_by_id('room-actions-btn').click()
-            self.context.driver.find_element_by_css_selector('.delete-room-action').click()
-            self.context.wait.until(EC.visibility_of_element_located((By.XPATH, '//button[text()="Delete room"]')))
-            self.context.driver.find_element_by_xpath('//button[text()="Delete room"]').click()
-            time.sleep(1)
-            # but we will know about that just when this will in master
+        self.context.driver.get(self.context.base_url + "/chat/room/" + url)
+        self.context.wait.until(lambda driver: driver.find_element_by_id('status_dropdown'))
+        self.context.driver.find_element_by_id('room-actions-btn').click()
+        self.context.driver.find_element_by_css_selector('.delete-room-action').click()
```

Comment:
Maybe we can do separated func for click "room-actions-btn" ? As I see, we sometimes use it in other functions.

Reply:
This good idea, I will do this way. Thank you for review!